workflow/src/main/kotlin/com/r3/corda/sdk/token/workflow/selection/TokenSelection.kt

```diff
@@ -188,10 +188,10 @@ class TokenSelection(val services: ServiceHub, private val maxRetries: Int = 8,
             // TODO This approach is clearly taken from core. Revisit it.
             services.keyManagementService.freshKeyAndCert(nodeIdentity, revocationEnabled = false).party.anonymise()
         } else {
-            // Check that this identity belongs to the node that called generateMove
+            // Check that this identity belongs to the node that called generateMove.
```

Comment:
right so... if no key is provided then a new confidential key is created for the node doing token selection. Otherwise, if there is a key, then we check that the key belongs to the node doing token selection (so we don't send change to someone else!).